apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/monetary/service/AssetDeleteServiceImpl.java

```diff
@@ -0,0 +1,88 @@
+/*
+ *  Copyright © 2018-2020 Apollo Foundation
+ */
+
+package com.apollocurrency.aplwallet.apl.core.monetary.service;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import java.util.stream.Stream;
+
+import com.apollocurrency.aplwallet.apl.core.app.Transaction;
+import com.apollocurrency.aplwallet.apl.core.converter.IteratorToStreamConverter;
+import com.apollocurrency.aplwallet.apl.core.db.DbClause;
+import com.apollocurrency.aplwallet.apl.core.db.DbIterator;
+import com.apollocurrency.aplwallet.apl.core.db.service.BlockChainInfoService;
+import com.apollocurrency.aplwallet.apl.core.monetary.model.AssetDelete;
+import com.apollocurrency.aplwallet.apl.core.monetary.dao.AssetDeleteTable;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Singleton
+public class AssetDeleteServiceImpl implements AssetDeleteService {
+
+    private final AssetDeleteTable assetDeleteTable;
+    private final BlockChainInfoService blockChainInfoService;
+    private IteratorToStreamConverter<AssetDelete> assetDeleteIteratorToStreamConverter =
+        new IteratorToStreamConverter<>();
+
+    @Inject
+    public AssetDeleteServiceImpl(AssetDeleteTable assetDeleteTable,
+                                  BlockChainInfoService blockChainInfoService,
+                                  IteratorToStreamConverter<AssetDelete> assetDeleteIteratorToStreamConverter) {
+        this.assetDeleteTable = assetDeleteTable;
+        this.blockChainInfoService = blockChainInfoService;
+        if (assetDeleteIteratorToStreamConverter != null) { // for unit test only
+            this.assetDeleteIteratorToStreamConverter = assetDeleteIteratorToStreamConverter;
+        }
+    }
+
+    @Override
+    public DbIterator<AssetDelete> getAssetDeletes(long assetId, int from, int to) {
```

Comment:
DbIterator should not be exposed to service/controller level

Reply:
Now we do not rewrite old REST api handlers, so we need those 'deprecated' methods for using by old REST api handlers. Otherwise we have to rewrite them too.